src/main/java/net/mcreator/ui/views/editor/image/tool/tools/AbstractTool.java

```diff
@@ -52,7 +58,7 @@
 	private JToggleButton toolPanelButton;
 
 	public AbstractTool(String name, String description, ImageIcon icon, Canvas canvas, ColorSelector colorSelector,
-			VersionManager versionManager) {
+			VersionManager versionManager, @Nonnull String opacitySliderName) {

```

Comment:
Why pass the opacity slider as a a string to the constructor of abstract tools? Abstract classes are meant to provide a base template for classes that extend them. In this case it's a base for tools and as you might have noticed yourself, tools don't always have opacity, so having it in the base tool is not the best way of doing this.

Reply:
:heavy_check_mark: Fixed
All related conversations will be resolved.